code/modules/projectiles/gun_system.dm

```diff
@@ -832,6 +832,8 @@
 		if(length(chamber_items) && CHECK_BITFIELD(reciever_flags, AMMO_RECIEVER_AUTO_EJECT) && CHECK_BITFIELD(reciever_flags, AMMO_RECIEVER_MAGAZINES) && get_current_rounds(chamber_items[current_chamber_position]) < (!CHECK_BITFIELD(reciever_flags, AMMO_RECIEVER_REQUIRES_UNIQUE_ACTION) ? rounds_per_shot : 0))
 			playsound(src, empty_sound, 25, 1)
 			unload(after_fire = TRUE)
+		if(length(chamber_items) && CHECK_BITFIELD(reciever_flags, AMMO_RECIEVER_CLEAR_CHAMBER_AFTER_SHOT))
+			QDEL_LIST(chamber_items)
```

Comment:
Why's it qdel'ing?

Wouldn't it just be better to have a flag or something that prevent auto eject toggling? Its not just the RL issue caused by that.

Reply:
Ehhh the disposable RL keeps the rocket it fired in it's chamber so. Probably